### PR TITLE
Add support for refresh_token

### LINKED ISSUE #15: Add support for refreshing tokens
As described in https://github.com/18F/cg-django-uaa/issues/22, cloud.gov has changed the expiry of access tokens to make them quite short-lived (currently 10-15 minutes), for security reasons.  This means that clients will definitely want to refresh tokens, so the fake cloud.gov ought to support it.

This functionality was added to the python version of the fake server in https://github.com/18F/cg-django-uaa/pull/24, so adding it to this go version should be pretty straightforward. We might also want to improve the example app so it auto-refreshes tokens too.

### PR DESCRIPTION
**Pull request by toolness**
This fixes #15.

To do:

- [x] Make the example client refresh tokens.
- [x] Make `TestRefreshAccessTokenWorks` and `TestExchangeCodeForAccessTokenWorks` examine the response and verify it looks OK.

### CODE REVIEW

**Comment by toolness** [comment]
@jseppi can you take a quick look at the changed `example-client.js` in this PR? does it make sense?

File: example-client/example-client.js [hunk 1]
```diff
@@ -26,27 +26,87 @@ const FAKE_STATE = crypto.randomBytes(64).toString('hex');
...
-// Our "home page" contains a login link.
+// Our "home page".
 app.get('/', (req, res) => {
-  const url = UAA_AUTH_URL + '?' + querystring.stringify({
-    'client_id': CLIENT_ID,
-    'state': FAKE_STATE,
-    'response_type': 'code'
-  });
-  res.send(`<a href="${url}">Log in</a> via <code>${UAA_AUTH_URL}</code>!`);
+  function getLoggedInHtml() {
```
Comment (jseppi) [inline example-client/example-client.js]: Very minor: I found it a little hard to parse what exactly the route handler was doing and think it would be clearer if just the `res.send(session.email ? getLoggedInHtml() : getLoggedOutHtml());` was in the handler and the two functions were moved outside.

File: example-client/example-client.js [hunk 2]
```diff
@@ -26,27 +26,87 @@ const FAKE_STATE = crypto.randomBytes(64).toString('hex');
...
+      grant_type: 'refresh_token',
+      refresh_token: session.refreshToken,
       client_id: CLIENT_ID,
       client_secret: CLIENT_SECRET,
-    },
-  }, callback);
-}
+    }, (err) => {
+      if (err) {
+        console.log('Renewal unsuccessful, user was logged out.');
```
Comment (jseppi) [inline example-client/example-client.js]: I think it would be slightly clearer if the setting of `session` was done here in this middleware function instead of up in the `postToTokenUrl` (similarly for the other use of `postToTokenUrl` down in `/auth/callback`).

My confusion was because as I was reading the code I got to this `console.log` and was curious _how_ the user was logged out. I had to refer back up to `postToTokenUrl` to see that session was set to `{}` in the case of `err`.

Comment (jseppi) [reply example-client/example-client.js]: Hrm, perhaps just changing the name of `postToTokenUrl` would be sufficient to making it clearer. Maybe something like `exchangeTokenAndSetSession` (blerf, that's pretty long, but you get the idea).

**Review by jseppi — APPROVED** [review]
Example looks good! Just made some minor comments about readability/clarity. They're just, like, my opinion, man, so take them or leave them :)

**Comment by toolness** [comment]
Thanks @jseppi !  I addressed all your suggestions in 03a55b11bc57448df48db8ee3f38ebdb5653a51d.

**Comment by toolness** [comment]
I think this looks good so imma merge.
